okhttp/src/main/java/okhttp3/internal/ws/WebSocketOptions.kt

```diff
@@ -0,0 +1,56 @@
+package okhttp3.internal.ws
+
+import okhttp3.Response
+import java.io.IOException
+
+data class WebSocketOptions(
+    @JvmField val compressionEnabled: Boolean,
+    @JvmField val contextTakeover: Boolean
+) {
+  companion object {
+    private const val OPTION_CLIENT_NO_CONTEXT_TAKEOVER = "client_no_context_takeover"
+    private const val OPTION_CLIENT_MAX_WINDOW_BITS = "client_max_window_bits"
+    private const val SUPPORTED_CLIENT_MAX_WINDOW_BITS = 15
+
+    private val DEFAULT = WebSocketOptions(
+        compressionEnabled = false,
+        contextTakeover = false
+    )
+
+    @Throws(IOException::class)
+    @JvmStatic
+    fun parseServerResponse(response: Response): WebSocketOptions =
+        response.header("Sec-WebSocket-Extensions")
+            ?.takeIf { it.contains("permessage-deflate") }
+            ?.let { extensionHeader ->
+              val options = extensionHeader.split(" ").map { it.replace(";", "") }
+
+              val clientNoContextTakeover = parseClientNoContextTakeover(options)
+              val clientMaxWindowBits = parseClientMaxWindowBits(options)
+
+              if (!clientNoContextTakeover) {
+                if (clientMaxWindowBits != null
```

Comment:
combine if blocks?

Reply:
Done. 
Actually, I think I should further improve it to be [closer to RFC,](https://tools.ietf.org/html/rfc7692#section-5.2) eg supporting quoted parameter values and fallback options.
Will do.

Reply:
Done